File: LeetCode_POTD/June_2026/June_13.py
"""
Simply done as stated in question with same steps:
Only optimization is:
    i) ans+=('z' - (sum % 26)) ->Better going with List because string ans need to completely first copy each char of previous ans and then add to assign new string(Immutable)

TC - O(Total chaacter), SC - O(1) {as output space is excluded}
"""

class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        ans=[]
        for s in words:
            sum=0
            for i in s:
                sum+=weights[ord(i)-ord('a')]
            ans.append(chr(ord('z')-(sum%26)))
        return ''.join(ans)

File: LeetCode_POTD/June_2026/June_13.java
package Leetcode_POTD.June_2026;

/*
Simply done as stated in question with same steps:
Only optimization is:
    i) ans+=('z' - (sum % 26)) ->Better going with StringBuilder because string ans need to completely first copy each char of previous ans and then add to assign new string(Immutable)
    ii)In the starting itself defining ans length

TC - O(Total chaacter), SC - O(1) {as output space is excluded}
*/

public class June_13 {
    public String mapWordWeights(String[] words, int[] weights) {
        StringBuilder ans = new StringBuilder(words.length);
        for (String word : words) {
            int sum = 0;
            for (int i = 0; i < word.length(); i++) {
                sum += weights[word.charAt(i) - 'a'];
            }
            ans.append((char)('z' - (sum % 26)));
        }
        return ans.toString();
    }
}
